lsp-semantic-tokens.el

```diff
@@ -230,15 +230,19 @@ If REGION is non-nil, it will request tokens only for given region
 otherwise it will request for whole document.
 If FONTIFY-IMMEDIATELY is non-nil, it will fontify when receive the response
 ignoring the timer."
-  (let ((request-full-token-set
-         (lambda (fontify-immediately)
-           (when lsp--semantic-tokens-idle-timer
-             (cancel-timer lsp--semantic-tokens-idle-timer))
-           (setq lsp--semantic-tokens-idle-timer
-                 (run-with-idle-timer
-                  lsp-idle-delay
-                  nil
-                  (lambda () (lsp--semantic-tokens-request nil fontify-immediately)))))))
+  (let* ((semantic-tokenizing-buffer (current-buffer))
+         (request-full-token-set
+          (lambda (fontify-immediately)
+            (when lsp--semantic-tokens-idle-timer
+              (cancel-timer lsp--semantic-tokens-idle-timer))
+            (setq lsp--semantic-tokens-idle-timer
+                  (run-with-idle-timer
+                   lsp-idle-delay
+                   nil
+                   (lambda ()
+                     (when (buffer-live-p semantic-tokenizing-buffer)
```

Comment:
this change looks reasonable to me, I'm actually doing the same in my pending semantic-tokens PR, so to the extent of my current knowledge I'd assume it to be correct :)